elements/pfe-tabs/pfe-tab-panel.ts

```diff
@@ -1,82 +1,23 @@
-import type { ColorTheme } from '@patternfly/pfe-core';
-
-import { LitElement, html } from 'lit';
 import { customElement, property } from 'lit/decorators.js';
 
-import { pfelement, initializer, colorContextConsumer } from '@patternfly/pfe-core/decorators.js';
 import { getRandomId } from '@patternfly/pfe-core/functions/random.js';
 
 import style from './pfe-tab-panel.scss';
-
-import { PfeTab } from './pfe-tab.js';
+import { BaseTabPanel } from './BaseTabPanel';
 
 /**
- * @slot - Add the content for your tab panel here.
+ * @slot - Tab panel content
  * @csspart container - container for the panel content
  */
-@customElement('pfe-tab-panel') @pfelement()
-export class PfeTabPanel extends LitElement {
+@customElement('pfe-tab-panel')
+export class PfeTabPanel extends BaseTabPanel {
   static readonly version = '{{version}}';
 
   static readonly styles = [style];
 
-  /**
-   * Sets color theme based on parent context
-   */
-  @colorContextConsumer()
-  @property({ reflect: true }) on?: ColorTheme;
-
-  /** Selected tab */
-  @property({ type: Boolean, reflect: true, attribute: 'aria-selected' }) selected = false;
-
-  /** Visibility */
-  @property({ type: Boolean, reflect: true }) hidden = false;
-
-  @property({ reflect: true }) role = 'tabpanel';
-
-  @property({ type: Number }) tabIndex = 0;
-
-  // TODO: Should deprecate
-  get tabindex() {
-    return this.tabIndex;
-  }
-
-  set tabindex(v: number) {
-    this.tabIndex = v;
-  }
-
-  @property({ attribute: 'aria-labelledby', reflect: true }) labelledby?: string;
-
-  /** Variant */
-  @property({ reflect: true }) variant: 'wind'|'earth' = 'wind';
-
-  connectedCallback() {
+  async connectedCallback() {
```

Comment:
does this need to be async?

Reply:
No it doesn't, moving `this.#upgradeAccessibility()` to firstUpdated().